### PR TITLE
Failed to load json_data to dataclass

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Fix errors in feast notebook.
Before
```
>>> print(feature_store)
>>> print(feature_store.config)

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)

/var/folders/by/42kc70lj1bb77klbhf2fmxx00000gn/T/ipykernel_46553/3923996928.py in <module>
      1 print(feature_store)
----> 2 print(feature_store.config)
      3 

~/opt/anaconda3/envs/flyte/lib/python3.8/site-packages/flytekit/core/type_engine.py in __getattr__(self, key)
    766                 return self.__getitem__(key)
    767             except KeyError:
--> 768                 raise AttributeError(key)
    769 
    770     return dataclass_json(dataclasses.dataclass(Model))

AttributeError: config
```
After 
https://github.com/pingsutw/flytesnacks/blob/interactive-feast/cookbook/case_studies/feature_engineering/feast_integration/Feast_Flyte_Demo.ipynb

The problem is that we didn't set the attribute for [model](https://github.com/flyteorg/flytekit/blob/7421be2e3608fce9a8ce25fb2077ce6dbb599e4e/flytekit/core/type_engine.py#L737-L756) class, so we can't load json_data to dataclass
https://github.com/flyteorg/flytekit/blob/7421be2e3608fce9a8ce25fb2077ce6dbb599e4e/flytekit/core/type_engine.py#L284

For example:
```
@dataclass_json
@dataclass()
class Bar(object):
    x: float
    y: str
```
We have to set attributes `x` and `y` for the model class when we guess data type. otherwise, TypeEngine will keep failing to translate Literal to python value

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by wild-endeavor** [comment]
cc @eapolinario 

This PR doesn't fully deprecate https://github.com/flyteorg/flytekit/pull/675/files right?  Should we merge that one in first?

For this PR, can we add a unit test that calls to `TypeEngine.dict_to_literal_map` like this https://github.com/flyteorg/flytekit/blob/7421be2e3608fce9a8ce25fb2077ce6dbb599e4e/tests/flytekit/unit/core/test_type_engine.py#L571

**Comment by eapolinario** [comment]
> This PR doesn't fully deprecate https://github.com/flyteorg/flytekit/pull/675/files right? Should we merge that one in first?

Correct, it doesn't, although we could merge this PR first and let https://github.com/flyteorg/flytekit/pull/675 be only about handling the inputs+outputs in flyteremote. What do you think, @wild-endeavor ?

**Comment by wild-endeavor** [comment]
yeah that sounds good.  let's still add that `dict_to_literal_map` unit test to this PR though, then +1

**Review by wild-endeavor — APPROVED** [review]
